[0.14 and Moby's Fishing Speed Chumpdate] Pet Changes

DuckySoLucky opened the pull request:
## Description 

Fixed #1581
Fixes #1606
Fixes #1604 / #1612
Part of #1617 

## Type of change
- [x] Updated feature

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

DuckySoLucky committed:
[0.14] Pet Changes

DuckySoLucky committed:
fix(wisdom symbol)

DuckySoLucky committed:
fix(random space idk :shrug:)

DuckySoLucky committed:
fear: add new accessories

nstringham commented on common/constants/stats.js:
```diff
@@ -181,6 +181,24 @@ export const statsData = {
+    suffix: "",
+    color: "b"
+  },
+  wisdom: {
```
interesting. I would have thought wisdom was already there

DuckySoLucky replied:
Same, not sure why it wasn't 🤷

dukio replied:
Wisdom is new, it was added to replace XXX skill xp boost

DuckySoLucky committed:
feat: add RGBEE Bee Pet skin

DuckySoLucky committed:
feat: add new symbols

metalcupcake5 commented:
accessories already added in #1608

DuckySoLucky commented:
Oops, my bad. I saw that pets weren't fully added so I thought accessories weren't done either

metalcupcake5 commented:
also it may have been a better idea to merge pet changes into #1612 but i can close that

DuckySoLucky commented:
Closing would be easier and faster
If you want I can merge but there isn't really need

DuckySoLucky committed:
fix: prettier check fail (maybe)

nstringham commented:
BTW @DuckySoLucky you can run the command
```bash
npm run prettier:fix
```
to fix any prettier errors

dukio commented:
If this PR solves the issues, please edit your first PR message to include
```
Fixes #1604
Fixes #1617
```

So those 3 issues get closed when this is merged 😄

dukio commented on common/constants/stats.js:
```diff
@@ -181,6 +181,78 @@ export const statsData = {
+    suffix: "",
+    color: "3",
+  },
+  rift_time: {
+    name: "Rift Time",
+    nameLore: "Rift Time",
+    nameShort: "Rift Time",
+    nameTiny: "RT",
+    symbol: "ф",
+    suffix: "",
+    color: "a",    
+  },
+  fear: {
+    name: "Fear",
+    nameLore: "Fear",
+    nameShort: "Fear",
+    nameTiny: "Fr",
+    symbol: "☠",
+    suffix: "",
+    color: "a",  
+  }
```
Oh even future stuff, nice lol

dukio commented on src/constants/accessories.js:
```diff
@@ -737,6 +737,21 @@ export const accessories = {
+    rarity: "legendary",
+    texture: "/head/9c7fb6f414ff0eb6bf5797acb3b3af354ca3af95ceb8f3418125f6b805419f39"
+  },
+  PANDORAS_BOX: {
+    name: "Pandora's Box",
+    rarity: "common",
+    texture: "/head/fe246925cbd4c05279b9c8dcf7c2fdfc9baf9b1424aa35e501533ebb98e00522",
+  },
```
For Pandora's Box we still need to figure out how to handle the fact that this is the only accessory that can't be recombobulated (so it shouldn't count in the total of accessories recombobulated)

dukio replied:
For Pandora's Box we still need to figure out how to handle the fact that this is the only accessory that can't be recombobulated (so it shouldn't count in the total of accessories recombobulated)

dukio commented on src/constants/skins-animations.js:
```diff
@@ -533,6 +533,12 @@ const skins = [
     texture: "/head/7cdb1ff8c967c831eb685e09e5dea586ed291472395acf7a25e9d5bb41d6c082",
     release: new Date("2022-07-08 18:00:00 GMT+1").getTime(),
   },
+  {
+    id: "PET_SKIN_BEE_RGBEE",
+    name: "RGBee",
+    texture: "/head/99c6da45da5a46614c05d5f8d7b3dfefb025e82ca4cdb4f0db64ced1315df659",
+    release: new Date("2022-08-26 18:00:00 GMT+1").getTime(),
+  },
```
This still needs the animation, I'll try to do it next week if I find some time

DuckySoLucky replied:
Okay, didn't know it was animated

dukio reviewed (commented):
lgtm, didn't test

added a few comments if you want to check those things out

DuckySoLucky committed:
fix: `npm run prettier:fix`, prettier errors

DuckySoLucky committed:
Wisp Pet Changes

DuckySoLucky committed:
fear: add Wisp Pet Changes

DuckySoLucky commented:
Fixed prettier errors, and updated wisp pet changes from #1581

MartinNemi03 commented:
> accessories already added in #1608

#1608 was merged now, so you can just pull that new version from development.

DuckySoLucky committed:
refactor: MartinNemi03's request

DuckySoLucky committed:
Merge branch 'development' into development

DuckySoLucky committed:
Updated to Changes from #1585

DuckySoLucky committed:
Merge branch 'development' of https://github.com/DuckySoLucky/SkyCrypt into development

DuckySoLucky committed:
refactor: Lint Variable Case

DuckySoLucky committed:
refactor: Lint Variable Case

DuckySoLucky committed:
fix: typo (petStats instead of PET_STATS)

I thought everything was moved to camelCase I guess not 🤷

dukio reviewed (approved):
lgtm

dukio commented:
> Fixes https://github.com/SkyCryptWebsite/SkyCrypt/issues/1617 (Animation is missing)

Just if you can remove this since by typing that it will close the issue while in this case the issue should remain open, I think you can type "Part of #1617" and it will not close the issue when this PR is merged.

DuckySoLucky commented:
### [September 6] Moby's Fishing Speed Chumpdate
> https://hypixel.net/threads/september-6-mobys-fishing-speed-chumpdate.5103151/

- [x] Ammonite
   - The Ammonite pet has received 2 changes to its perk.
     First, "Not a Snail" speed and defense have been moved to Gift of the Ammonite.
     Second, "Not a Snail" has been replaced with "Expert Cave Fisher", which makes the Ammonite useful for underground fishing situations.

- [x] Dolphin
   - Pod Tactics from Dolphin pet 9->10 Fishing Speed and 10->30 blocks range.

- [x] Jellyfish
   - Moved the Health Regen from the Jellyfish Pet's Radiant Regeneration to a base stat on the pet and renamed the perk to Radiant Scyphozoa.

DuckySoLucky committed:
fix: [0.14] Pet Changes

DuckySoLucky committed:
fix (jellyFish health regen stats)

DuckySoLucky committed:
fix (typo)

DuckySoLucky committed:
fix

DuckySoLucky committed:
refactor: changed `&` to `§`

DuckySoLucky commented:
Tested
![image](https://user-images.githubusercontent.com/75372052/189483253-9f9aa35d-45e8-4d44-9502-083654487273.png)
![image](https://user-images.githubusercontent.com/75372052/189483459-9b1036d7-9255-47ef-86fc-20add35b8f23.png)
![image](https://user-images.githubusercontent.com/75372052/189483471-6771790f-5aa6-431c-a07d-798e0313eadb.png)
![image](https://user-images.githubusercontent.com/75372052/189483491-33c580b9-8511-4ce7-968e-85fd8cce831b.png)
![image](https://user-images.githubusercontent.com/75372052/189483497-49c09e93-447d-4911-b8b6-01e0466ec697.png)
![image](https://user-images.githubusercontent.com/75372052/189483515-8b18e6aa-fed4-46af-a1a1-233997f0b8b4.png)
![image](https://user-images.githubusercontent.com/75372052/189483522-83b7d614-7c8e-46db-8940-557b4b80106f.png)

DuckySoLucky committed:
fix (typo)

dukio commented on src/constants/pet-stats.js:
```diff
@@ -2642,16 +2667,10 @@ class Jellyfish extends Pet {
   get first() {
-    const multHealth = getValue(this.rarity, { epic: 1 });
-    const multMana = getValue(this.rarity, { epic: 0.5 });
+    const mult_mana = getValue(this.rarity, { epic: 0.5 });
```
If you can fix  this

> GitHub Actions
/ es-lint
Variable name `mult_mana` must match one of the following formats: camelCase, UPPER_CASE     

so `multMana`

dukio commented on src/constants/pet-stats.js:
```diff
@@ -1448,16 +1467,16 @@ class Phoenix extends Pet {
 
   get first() {
-    const startStrength = getValue(this.rarity, { epic: 10, legendary: 15 });
+    const start_strength = getValue(this.rarity, { epic: 10, legendary: 15 });
```
This one too 
> Variable name `start_strength` must match one of the following formats: camelCase, UPPER_CASE

so `startStrength`

metalcupcake5 commented on common/constants/stats.js:
```diff
@@ -181,6 +181,78 @@ export const STATS_DATA = {
+    symbol: "☠",
+    suffix: "",
+    color: "a",
+  },
```
imo these shouldn't be added until they are added in game

metalcupcake5 reviewed (approved):
didnt test

DuckySoLucky committed:
style: camelCase

DuckySoLucky committed:
style: remove useless stats (not in game yet)

dukio reviewed (approved):
lgtm, tested (didn't check individual values for pet stats but I trust they are correct)

dukio commented:
if anyone else wants to test this PR, you need to upstream it otherwise you would get the CONSTATS is not defined error (already fixed in another merged PR)

dukio commented on common/constants/stats.js:
```diff
@@ -181,6 +181,60 @@ export const STATS_DATA = {
+    suffix: "",
+    color: "a",
+  },
+  wisdom: {
+    name: "Wisdom",
+    nameLore: "Wisdom",
+    nameShort: "Wisdom",
+    nameTiny: "W",
+    symbol: "☯",
+    suffix: "",
+    color: "3",
+  },
```
Actually can you add a comment to this since ig it's just temporary until #1639 it's merged (with all the real wisdoms)?

like:
`// todo: remove this once #1639 is merged and update wisdom values`

DuckySoLucky committed:
Update stats.js

MartinNemi03 reviewed (approved):
lgtm!
